File: 프로그래머스/0/120902. 문자열 계산하기/문자열 계산하기.js
function solution(my_string) {
    const stack = [];
    let sign = 1;
    
    for (const v of my_string.split(" ")) {
        if (v === "+") {
            sign = 1;
        } else if (v === "-") {
            sign = -1;
        } else {
            stack.push(v * sign);
        }
    }

    return stack.reduce((a,b) => a + b, 0);
}

File: 프로그래머스/0/120902. 문자열 계산하기/문자열 계산하기.py
def solution(my_string):
    strArr = my_string.split()
    answer = int(strArr[0])
    for i in range(1,len(strArr),2):
        if strArr[i] == '+':
            answer += int(strArr[i+1])
        else:
            answer -= int(strArr[i+1])
    return answer